src/main/kotlin/sipura/generate/Factory.kt

```diff
@@ -78,4 +78,36 @@ object Factory {
             }
         }
     }
+
+    /**
+     * Creates a split graph with an independent set of size [sizeIS] and a clique of size [sizeC]. Also adds all
+     * possible edges between the independent set and the clique.
+     *
+     * The vertices of the independent set are [1, ..., [sizeIS]] and the vertices of the clique
+     * are [[sizeIS] + 1, ..., [sizeIS] + [sizeC]].
+     *
+     * @throws IllegalArgumentException if either [sizeIS] or [sizeC] are not positive.
+     * @return The created split graph.
+     */
+    fun createSplitGraph(sizeIS: Int, sizeC: Int): SimpleGraph<Int> {
+        if (sizeIS < 0 || sizeC < 0) throw IllegalArgumentException("Both sides must have non-negative size")
```

Comment:
0 is not positive, but no exception is thrown for it.
We have to make this consistent, even though I'm not sure whether 0 should be a valid input.

Reply:
Ok from the unit tests it seems like size 0 is fine for both parameters. IMO we can keep it like that

Reply:
done